examples/step-79/doc/intro.dox

```diff
@@ -0,0 +1,597 @@
+smaller values. At larger values, it is multiplied by a constant (around .6),
+and at lower values the barrier value is replaced by the barrier value raised
+to some exponent (around 1.2). This method has proven to be effective at keeping
+ the subproblem solvable at large barrier values, while still allowing
+ superlinear convergence at smaller barrier values.
+@f[
+  \alpha_{k+1} = \min\{\alpha_k^{1.2},.6\alpha_k\}
+@f]
+
+While taking large steps at reducing the barrier size when convergence is
+reached is widely used, more recent research has shown that it typically faster
+to use algorithms that adaptively update barrier each iteration, i.e., in which
+we use concrete criteria at the end of each iteration to determine what the
+penalty parameter should be in the next iteration, rather than using reduction
+factors that are independent of the current solution. That said, such methods
+are also more complicated and we will not do this here.
+
+<h3>Merit Function</h3>
+
+The algorithm outlined above makes use of a "merit function". Merit functions
+are used to determine whether a step from $x_k$ to a proposed point $x_{k+1}$ is
+ beneficial. In unconstrained optimization problems, one can simply check this
+ with the object function we try to minimize, and typically uses conditions such
+  as the Wolfe and Goldstein conditions.
+
+In constrained optimization problems, the question is how to balance reduction
+in the objective function against a possible increase in the violation of
+constraints: A proposed step might make the objective function smaller but be
+further away from the set of points that satisfy the constraints -- or the other
+ way around. This trade-off is typically resolved by using a merit function that
+  combines the two criteria.
+
+Here, we use an exact $l_1$ merit function to test the steps.
+@f{multiline}{
+  \phi(\textbf{x},\textbf{y}) = \int_{\partial \Omega} \textbf{u}\cdot
+  \textbf{t} - \alpha \int_\Omega (log(s_1) + log(s_2)) + p \sum_i\left|
+  \int_\Omega y_{2,i}(H(\sigma) - \rho) \right|
+\\
++ p \sum_i\left| \int_{\partial \Omega} \textbf{y}_{1,i}\cdot \textbf{t}  -
+\int_\Omega \rho^p[\lambda \nabla \cdot \textbf{u} \nabla \cdot \textbf{y}_{1,i}
+ + \mu \varepsilon{\textbf{u}}\varepsilon{\textbf{y}_{1,i}}]\right|
+\\
++ p \sum_i\left| \int_\Omega z_{1,i}(s_1 - \sigma) \right|+ p \sum_i\left|
+\int_\Omega z_{2,i}(1-\sigma - s_2) \right|
+@f}
+
+Here, $p$ is a penalty parameter. This merit function being exact means that
+there exists some $p_0$ so that for any $p > p_0$, the merit function has its
+minima at the same location as the original problem. This penalty parameter is
+updated (by recommendation of Nocedal and Wright @cite Benson2002) as follows:
+@f[
+  p > \frac{.5 x^T H x - x^t\nabla f}{\norm{c_i}_{l_\infty}, i \in \mathcal{E}}
+@f]
+Where $H$ is the hessian of the objective function, $x$ is a vector of our
+decision variables, $f$ is the objective function, and $c_i$ is the error on a
```

Comment:
You've not used the term "decision variables" before. What does this mean?

Reply:
Decision variables is a fairly common term in linear/nonlinear programming for the variables that can be changed in the objective function. Because I'm using log barriers, the slack variables are a part of the barrier function that I treat as the objective function, so it's actually exactly all of my primal variables here - I can make that change pretty easily.

Reply:
Decision variables is typically a way of talking about the variables in the objective function, so primal not-slack variables. That said, because I am actually using log barriers, I replace the objective function a barrier function that does involve the slacks, so in the case of this problem it is all of the primal variables. Decision variables is more technically correct here as it translates better to other problems, but I could easily replace this with primal variables and have it still be correct.

Reply:
Then it sounds perfectly reasonable to want to keep this terminology in place. But could you then write it as "decision (primal) variables" or "primal (decision) variables"? Since you introduce it as "our decision variables", one gets the feeling that one should be familiar with the term after reading the preceding paragraphs (which would not be the case for readers not familiar with topology optimisation).